### ISSUE TITLE
Aws\recursive_dir_iterator doubles slashes in response items

### ISSUE THREAD

**Issue by TheRoSS** [aws/aws-sdk-php#1109]
When I use '/' at the end of the $path parameter (this is reasonable because this is a directory and other API calls require slash at the end of path) I have double slashes in the response items. This is very confusing behaviour. 

Furthermore, the recursive_dir_iterator call with '/' at the end of path parameter is fulfiled much longer as compared with the call without '/'.

Example:
```php
$items = Aws\recursive_dir_iterator("s3://myBacket/dir/");
foreach ($items as $key) {
    echo("KEY: $key\n");
}
```
will give:
```
KEY: s3://myBacket/dir//file1.txt
KEY: s3://myBacket/dir//file2.txt
KEY: s3://myBacket/dir//file3.txt
```

I use the 3.18.32 version of SDK, but the current version 3.19.17 has this bug too.

To resolve this issue you need something like this:
file - src/functions.php:214
replace
```php
$fullPath = "{$path}/{$file}";
```
with
```php
$fullPath = $path[$pathLen-2] == "/" ? "{$path}{$file}" : "{$path}/{$file}";
```

**Comment by cjyclaire** [comment]
@TheRoSS Thanks for the feedback and noticing this behavior, apology for the confusion there. 
I understand that ends a directory with '/' appears normal behavior, however, in S3 convention(from web console) and all of SDK examples, you will notice there is also no end '/', in addition, php [`__DIR__`](http://php.net/manual/en/language.constants.predefined.php) also has no '/' in ending etc. 

It make sense to me that the extra '/' in the filename needs to be removed, yet the way you fix it will change the behavior with current SDK. For example, given directory `s3://bucket/test`, if there is also a `s3://bucket/test/sub-test`. Result for using `Aws\recursive_dir_iterator("s3://bucket/test/")` will list only the first level items, and `Aws\recursive_dir_iterator("s3://bucket/test")` will list every thing in side `test` recursively. Changing the behavior there might break other customer depending this function. 

In summary, it's generally a good practice to omit the extra '/' there, having '//' in result serves as an indicator for removing the extra '/' there. Appreciate your thoughts on this, if you still prefer no double '/' exist in result, I could open a PR to fix it from result instead of fixing it in generating process. Thoughts?

**Comment by cjyclaire** [comment]
Closing due to ages, yet feel free to reopen with further questions or comments :)
